torchvision/datasets/forksafeziplookup.py

```diff
@@ -0,0 +1,46 @@
+import zipfile
+import threading
+import os, io
+import struct
+
+
+_FH_FILENAME_LENGTH = 10
+_FH_EXTRA_FIELD_LENGTH = 11
+STRUCT_FILE_HEADER = "<4s2B4HL2L2H"
+SIZE_FILE_HEADER = struct.calcsize(STRUCT_FILE_HEADER)
+
+
+# thread-safe/multiprocessing-safe
+class ForkSafeZipLookup:
+    def __init__(self, filename):
+        self.root_zip_filename = filename
+        self.root_zip_lookup = {}
+        self.root_zip = {}
+
+        with zipfile.ZipFile(filename, "r") as root_zip:
+            for info in root_zip.infolist():
+                if info.is_dir():
+                    continue
+                if info.compress_type != zipfile.ZIP_STORED:
+                    raise ValueError("Only uncompressed ZIP file supported: " + info.filename)
+                if info.compress_size != info.file_size:
+                    raise ValueError("Must be the same when uncompressed")
+                self.root_zip_lookup[info.filename] = (info.header_offset, info.compress_size)
+
+
+    def __getitem__(self, path):
+        key = (os.getpid(), threading.get_ident())
```

Comment:
From the documentation of `threading.get_ident`
> A thread's identity may be reused for another thread after it exits.

So is this safe to do across epochs? Because we create new processes at the beginning of every epoch.

Reply:
Combined with getpid(), it is definitely safe (on Linux you get a different pid for every thread).

Reply:
This code is removed.